### PR TITLE
Fixes an issue with type coercion

### PR DESCRIPTION
**Pull request by eliias**
Attempt to fix https://github.com/matsadler/magnus/issues/21

### CODE REVIEW

**Comment by matsadler** [comment]
Hey, thanks for this!

I had thought this would be ok as I checked the docs and [`CString::into_raw`](https://doc.rust-lang.org/stable/std/ffi/struct.CString.html#method.into_raw) is documented as returning `i8`. Turns out I didn’t look quite hard enough as [the source](https://doc.rust-lang.org/stable/src/alloc/ffi/c_str.rs.html#450) uses the `c_char` alias which resolves to `u8`/`i8` depending on the platform (and I guess rustdoc 'helpfully' resolves for display too).

So this should probably use `c_char`.

**Comment by eliias** [comment]
@matsadler thanks for taking a look. I think the docs confirm your finding as well https://doc.rust-lang.org/stable/core/ffi/type.c_char.html

File: src/error.rs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 //! Rust types for working with Ruby Exceptions and other interrupts.
 
+use core::ffi::c_char;
 use std::{any::Any, borrow::Cow, ffi::CString, fmt, mem::transmute, ops::Deref, os::raw::c_int};
```
Comment (matsadler) [inline src/error.rs:4]: Could you add `c_char` to the end of the `use std::…` line, like `…, os::raw::{c_char, c_int}};` instead.

`core::ffi::c_char` was nightly only up until the latest release of Rust, whereas `std::os::raw::c_char` has been available for ages (even though it’s just a reexport of `core::ffi::c_char`), and I’d like to keep things working on older Rust versions.

Comment (eliias) [reply src/error.rs]: Fixed.

**Pull request merged**